Trademark guidelines update (#4030)
jenkins-infra/jenkins.io: content/project/governance.adoc

The Jenkins project as a community revolves around Jenkins as a piece of software. We are a group of open-source developers and users who develop, use, promote Jenkins, software around Jenkins, and other related activities for our mutual benefit.

The project is a part of the https://cd.foundation/[Continuous Delivery Foundation] and

> affiliated with the link:https://www.linuxfoundation.org/[Linux Foundation] legal entities which legally own the project assets and acts a treasury.

Reviewer: Depending on intended meaning,

```suggestion
affiliated with the link:https://www.linuxfoundation.org/[Linux Foundation] legal entities which legally owns the project assets and acts as treasury.
```

(still ambiguous)

or

```suggestion
affiliated with the link:https://www.linuxfoundation.org/[Linux Foundation] legal entities which legally own the project assets and act as treasury.
```

> affiliated with the link:https://www.linuxfoundation.org/[Linux Foundation] legal entities which legally owns the project assets and acts as treasury.

Reviewer: ```suggestion
affiliated with the link:https://www.linuxfoundation.org/[Linux Foundation] legal entities which legally own the project assets and act as treasury.
```

Entities is plural (unless it's a weird legal term or something).

Before 2021, these roles were held by 

> link:https://www.linuxfoundation.org/[Software in the Public Interest] (SPI).

Reviewer: ```suggestion
link:https://www.spi-inc.org/[Software in the Public Interest, Inc.] (SPI).
```
